File: nomad-samples/common/js/sync-1.1/movie/get-movies.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Get all movies
 *
 * @param {string} authToken    | The authorization token
 *
 * @returns {Array} List of search results
 */
export default async function getMovies(authToken) {
    // Check for valid parameters
    if (!authToken) {
        throw new Error("Get Movies: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${authToken}`);

    // Build the payload body
    const body = {
        filters: [
            {
                fieldName: "contentDefinitionId",
                operator: "Equals",
                values: prjConstants.MOVIE_CONTENT_DEFINITION_ID
            },
            {
                fieldName: "languageId",
                operator: "Equals",
                values: prjConstants.US_ENGLISH_LANGUAGE_LOOKUP_ID
            }
        ],
        SearchResultFields: [
            {
                name: "contentDefinitionId"
            },
            {
                name: "contentDefinitionTitle"
            },
            {
                name: "plot"
            },
            {
                name: "releaseDate"
            },
            {
                name: "genre"
            },
            {
                name: "image",
                SearchResultFields: [
                    {
                        name: "fullUrl"
                    }
                ]
            }
        ]
    };

    // Send POST request
    const response = await fetch(`${prjConstants.ADMIN_API_URL}/admin/search`, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(body)
    });

    // Check for success
    if (response && response.ok) {
        // Get the response
        const movies = await response.json();

        // Return the movies
        return movies;
    }

    await apiExceptionHandler(response, "Get Movies failed");
}

File: nomad-samples/admin/py/sync-1.1/movies/get_movies.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests

def get_movies(AUTH_TOKEN):

    HEADERS = {
        "Authorization": "Bearer " +  AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    BODY = {
        "filters": [
            {
                "fieldName": "contentDefinitionId",
                "operator": "Equals",
                "values": MOVIE_CONTENT_DEFINITION_ID
            },
            {
                "fieldName": "languageId",
                "operator": "Equals",
                "values": US_ENGLISH_LANGUAGE_LOOKUP_ID
            }
        ],
        "SearchResultFields": [
            {
                "name": "contentDefinitionId"
            },
            {
                "name": "contentDefinitionTitle"
            },
            {
                "name": "plot"
            },
            {
                "name": "releaseDate"
            },
            {
                "name": "genre"
            },
            {
                "name": "image",
                "SearchResultFields": [
                    {
                        "name": "fullUrl"
                    }
                ]
            },
            {
                "name": "movieFile",
                "SearchResultFields": [
                    {
                        "name": "fullUrl"
                    }
                ]
            }
        ]
    }

    try:
        RESPONSE = requests.get(f"{ADMIN_API_URL}/admin/search", headers= HEADERS, data= json.dumps(BODY))

        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
        
        return RESPONSE.json()
    
    except:
        api_exception_handler(RESPONSE, f"Getting Movies failed")